Signal consistency check.

WojciechRynczuk opened the pull request:
It must be detected when user wants to add two signals with the same names but different properties (type/size/source).

The name does not have to be verified as when it is different the signal is different even if the remaining properties are the same.

endrjux commented on sources/common/src/SignalDb.cpp:
```diff
@@ -50,6 +50,7 @@ void SIGNAL::SignalDb::Add(const SIGNAL::Signal *pSignal)
     // A signal shall have a valid source once added to the database.
     if (pSignal->GetSource() == SourceRegistry::BAD_HANDLE)
     {
+        delete pSignal;
```
Why? This function didn't allocate it.

WojciechRynczuk replied:
Good catch.

endrjux commented on sources/common/src/TxtParser.cpp:
```diff
@@ -60,15 +61,31 @@ void PARSER::TxtParser::Parse()
+            }
+            catch (EXCEPTION::VcdException &rException)
+            {
+                if (EXCEPTION::Error::INCONSISTENT_SIGNAL == rException.GetId())
```
`delete` should be here. We should consider using `std::unique_ptr` to get rid of such problems.

WojciechRynczuk replied:
Yes, I thought of it as well. It seems it would be a good opportunity to do it when we will be refactoring the database of signals to save some memory. Well, in the next release.

endrjux commented:
I see new commit, but in "changes" windows I still see old diffs. Why?

WojciechRynczuk commented:
I have no idea why it is shown this way. In the "changes" window I have to select the specific commit to see the changes. Interestingly it also says 'no changes since the last review' which isn't correct. Anyway, the file has been changed and the right branch has been chosen.
